cursorless-talon/src/modifiers/scopes.py

```diff
@@ -13,60 +32,49 @@
     desc="Supported plural custom regular expression scope types",
 )
 
-
-@mod.capture(
-    rule="{user.cursorless_scope_type}"
-    " | <user.cursorless_surrounding_pair_scope_type>"
-    " | <user.cursorless_glyph_scope_type>"
-    " | {user.cursorless_custom_regex_scope_type}"
+mod.list(
+    "cursorless_scope_type_flattened",
+    desc="All supported scope types flattened",
+)
+mod.list(
+    "cursorless_scope_type_flattened_plural",
+    desc="All supported plural scope types flattened",
 )
-def cursorless_scope_type(m) -> dict[str, str]:
-    """Cursorless scope type singular"""
-    try:
-        return {"type": m.cursorless_scope_type}
-    except AttributeError:
-        pass
-
-    try:
-        return m.cursorless_surrounding_pair_scope_type
-    except AttributeError:
-        pass
 
-    try:
-        return m.cursorless_glyph_scope_type
-    except AttributeError:
-        pass
 
-    return {
-        "type": "customRegex",
-        "regex": m.cursorless_custom_regex_scope_type,
-    }
+@mod.capture(rule="{user.cursorless_scope_type_flattened}")
```

Comment:
This is the change where the performance increase actually comes from. Instead of referencing multiple captures and list we now just have a single list.